### ISSUE TITLE
nice_t_test for specific mean?

### ISSUE THREAD

**Issue by philpav** [rempsyc/rempsyc#2]
Hi, 

I want to combine multiple "two.sided" t-tests with a specific mean into a nice_table. But how can I use `nice_t_test() `to achieve this?
In base R I would write as this: 
`t.test(df$var, mu=10, alternative = "two.sided")`

Is it even possible?

Thank you!

**Comment by rempsyc** [comment]
Hi, thank you for opening an issue!

You are correct that in the current version, doing a one-sample t-test provides an error because the group argument is currently mandatory:

```r
nice_t_test(data = mtcars,
            response = "mpg",
            mu = 10)

Error in (function(x, i, exact) if (is.matrix(i)) as.matrix(x)[[i]] else .subset2(x, : 
  argument "group" is missing, with no default
```

With version `0.0.3.2`, it is now possible:
```r
nice_t_test(data = mtcars,
            response = "mpg",
            mu = 10)

Dependent Variable        t df                  p        d CI_lower CI_upper
               mpg 9.470995 31 0.0000000001154598 3.333466 2.472257 4.291719
```
We can confirm it yields identical results to `t.test`:
```r
t.test(mtcars$mpg, mu = 10)

	One Sample t-test
data:  mtcars$mpg
t = 9.471, df = 31, p-value = 0.0000000001155
alternative hypothesis: true mean is not equal to 10
```
For the sake of simplicity I have ignored the `alternative = "two.sided"` argument as it is the default (though feel free to specify it for clarity in your own code).

It can be used with several variables as well:
```r
nice_t_test(data = mtcars,
            response = names(mtcars)[1:7],
            mu = 10)
 
  Dependent Variable          t df                                           p        d CI_lower  CI_upper
1                mpg   9.470995 31 0.00000000011545976649382190319615901219485 3.333466 2.472257  4.291719
2                cyl -12.075982 31 0.00000000000029571269126768929345894232341 3.464598 2.575163  4.455521
3               disp  10.074267 31 0.00000000002684882863765266898178896148863 1.861581 1.299482  2.471701
4                 hp  11.277551 31 0.00000000000168302581052159496506799574167 2.139460 1.524730  2.811460
5               drat -67.747809 31 0.00000000000000000000000000000000002932045 6.726586 5.101919  8.516288
6                 wt -39.213794 31 0.00000000000000000000000000055204152227267 3.288084 2.436733  4.235071
7               qsec  24.846472 31 0.00000000000000000000049639366365302937307 9.988426 7.515941 12.616879
```
I have also updated the documentation. Thank you for your contribution in pointing out this issue.

**Comment by philpav** [comment]
Amazing!
Thank for the quick fix!

I have also tried with `alternative = "less" `and `alternative = "greater"` and it's also working. 

My workaround was to use the report package: 

```
model = t.test(mtcars$mpg, mu = 10)
tbl = as.data.frame(report::report(model))
nice_table(tbl, report = "t.test")
```

**Comment by philpav** [comment]
Hi!

I am afraid I found an issue. 

```
t.test(mtcars$mpg, mu=10)
nice_t_test(data = mtcars, response = "mpg", mu = 10, warning = FALSE)

t.test(mtcars$cyl, mu=10)
nice_t_test(data = mtcars, response = "cyl", mu = 10, warning = FALSE)
```
The confidence intervals differ.

**Comment by rempsyc** [comment]
There is an explanation for that. The confidence interval from `nice_t_test` is for the effect size, Cohen's _d_, whereas the one from `t.test` is for the _t_ value. Usually people are more interested in the effect size and its interval than the _t_ value.

**Comment by rempsyc** [comment]
I am closing this issue for now but don't hesitate to reopen it if I have not answered your question.

**Comment by rempsyc** [comment]
I am closing this issue for now but don't hesitate to reopen it if I have not answered your question.

**Comment by rempsyc** [comment]
I think I now better understand the issue you were describing. I have decided to compare the CI with that of the `report` package and noticed that they differed because the Cohen's _d_ actually differed.

Demo here:
effectsize::cohens_d(mtcars$mpg, mu=10)
Cohen's d |       95% CI
------------------------
1.67      | [1.15, 2.24]

effectsize::cohens_d(mtcars$cyl, mu=10)
Cohen's d |         95% CI
--------------------------
-2.13     | [-2.81, -1.52]


I believed that mu was passed automatically to the `effectsize::cohens_d()` function from the t.test object, but it was not.

**Comment by rempsyc** [comment]
I think I now better understand the issue you were describing. I have decided to compare the effect size CI with that of the `report` package and noticed that they differed because the Cohen's _d_ actually differed.

Demo here:
```r
effectsize::cohens_d(mtcars$mpg, mu=10)$Cohens_d
[1] 1.674251
rempsyc::nice_t_test(data = mtcars, response = "mpg", mu = 10, warning = FALSE)$d
[1] 3.333466

effectsize::cohens_d(mtcars$cyl, mu=10)$Cohens_d
[1] -2.134752
rempsyc::nice_t_test(data = mtcars, response = "cyl", mu = 10, warning = FALSE)$d
[1] 3.464598

```
I believed that `mu` was passed automatically to the `effectsize::cohens_d()` function from the `t.test` object, but it was not (so it used the default of `0`).

This is fixed in version `0.0.3.5`:
```r
effectsize::cohens_d(mtcars$mpg, mu=10)$Cohens_d
[1] 1.674251
rempsyc::nice_t_test(data = mtcars, response = "mpg", mu = 10, warning = FALSE)$d
 [1] 1.674251

effectsize::cohens_d(mtcars$cyl, mu=10)$Cohens_d
[1] -2.134752
rempsyc::nice_t_test(data = mtcars, response = "cyl", mu = 10, warning = FALSE)$d
 [1] -2.134752
```
Thanks for reporting the issue.

**Comment by rempsyc** [comment]
I am closing this issue for now but don't hesitate to reopen it if I have not answered your question.
